var/spack/repos/builtin/packages/zfp/package.py

```diff
@@ -62,53 +73,86 @@ class Zfp(CMakePackage, CudaPackage):
         "use smaller for finer rate granularity. "
         "Use 8 for H5Z-ZFP filter.",
     )
+
     variant(
-        "strided", default=False, description="Enable strided access for progressive zfp streams"
+        "daz",
+        default=False,
+        description="Denormals are zero: "
+        "Treat denormal-only blocks as containing "
+        "all zeroes",
     )
-    variant("aligned", default=False, description="Enable aligned memory allocation")
-    variant("twoway", default=False, description="Use two-way skew-associative cache")
+
     variant(
         "fasthash",
         default=False,
         description="Use a faster but more collision prone hash function",
     )
+
     variant("profile", default=False, description="Count cache misses")
 
+    variant(
+        "round",
+        default="never",
+        values=("never", "first", "last"),
+        multi=False,
+        description="EXPERIMENTAL: Set coefficient rounding method",
+    )
+
+    variant(
+        "strided", default=False, description="Enable strided access for progressive zfp streams"
+    )
+
+    variant(
+        "tight-error",
+        default=False,
+        description="EXPERIMENTAL: Use tighter error bound when rounding is enabled",
+    )
+
+    variant("twoway", default=False, description="Use two-way skew-associative cache")
+
     # Conflicts
     conflicts("+c", when="@:0.5.3", msg="+c requires zfp 0.5.4 or later")
-    conflicts("+python", when="@:0.5.4", msg="+python requires zfp 0.5.5 or later")
-    conflicts("+fortran", when="@:0.5.4", msg="+fortran requires zfp 0.5.5 or later")
-    conflicts("+openmp", when="@:0.5.2", msg="+openmp requires zfp 0.5.3 or later")
+    conflicts("+daz", when="@:0.5.5", msg="+daz requires zfp 1.0.0 or later")
+    conflicts("+python", when="@:0.5.4", msg="+python requires zfp 0.5.4 or later")
+    conflicts("+fortran", when="@:0.5.4", msg="+fortran requires zfp 0.5.4 or later")
+    conflicts("+openmp", when="@:0.5.2", msg="+openmp requires zfp 0.5.4 or later")
     conflicts("+cuda", when="@:0.5.3", msg="+cuda requires zfp 0.5.4 or later")
     conflicts("+fasthash", when="@:0.5.1", msg="+fasthash requires zfp 0.5.2 or later")
     conflicts("+profile", when="@:0.5.1", msg="+profile requires zfp 0.5.2 or later")
 
-    # Dependencies
-    depends_on("cmake@3.4.0:", type="build")
-    depends_on("cuda@7:", type=("build", "test", "run"), when="+cuda")
-    depends_on("python", type=("build", "test", "run"), when="+python")
-    depends_on("py-numpy", type=("build", "test", "run"), when="+python")
-    depends_on("py-cython", type="build", when="+python")
+    conflicts("round=never", when="@:0.5.5", msg="round requires zfp 1.0.0 or later")
+    conflicts("round=first", when="@:0.5.5", msg="round requires zfp 1.0.0 or later")
+    conflicts("round=last", when="@:0.5.5", msg="round requires zfp 1.0.0 or later")
```

Comment:
An alternative to using `conflicts` for the values -- which will trigger an error that shows the `msg` -- is to add a `when` option to the `variant` itself to indicate when the variant applies (e.g., `when="1.0.0:").

Reply:
Ah, that seems like a better approach.  Thanks for the suggestion.

Reply:
👍 I regret not mentioning it first.

Reply:
I've added this change with the latest fix. Let me know if you see any other issues.